model_compression_toolkit/gptq/pytorch/quantizer/ste_rounding/symmetric_ste.py

```diff
@@ -0,0 +1,200 @@
+                quantization_method=[QuantizationMethod.POWER_OF_TWO, QuantizationMethod.SYMMETRIC],
+                quantizer_type=RoundingType.STE)
+class STEWeightQuantizer(BasePytorchGPTQTrainableQuantizer):
+    """
+    Trainable symmetric quantizer to quantize a layer weights.
+    """
+
+    def __init__(self,
+                 quantization_config: TrainableQuantizerWeightsConfig,
+                 max_lsbs_change_map: dict = DefaultDict({}, lambda: 1)):
+        """
+        Construct a Pytorch model that utilize a fake weight quantizer of STE (Straight Through Estimator) for symmetric quantizer.
+
+        Args:
+            quantization_config: Trainable weights quantizer config.
+        """
+        super().__init__(quantization_config)
+        self.num_bits = quantization_config.weights_n_bits
+        self.per_channel = quantization_config.weights_per_channel_threshold
+
+        threshold_values = quantization_config.weights_quantization_params[THRESHOLD]
+        self.threshold_shape = np.asarray(threshold_values).shape
+        self.threshold_values = np.reshape(np.asarray(threshold_values), [-1]) if self.per_channel else float(
+            threshold_values)
+
+        self.quantization_axis = quantization_config.weights_channels_axis
+        self.power_of_two = quantization_config.weights_quantization_method == QuantizationMethod.POWER_OF_TWO
+        self.max_lsbs_change = max_lsbs_change_map.get(self.num_bits)
+        self.quantizer_parameters = {}
+
+
+    def initialize_quantization(self,
+                                tensor_shape: torch.Size,
+                                name: str,
+                                layer: qi.PytorchQuantizationWrapper) -> Dict[str, nn.Parameter]:
+        """
+        Return a dictionary of quantizer parameters and their names.
+
+        Args:
+            tensor_shape: tensor shape of the quantized tensor.
+            name: Tensor name.
+            layer: Layer to quantize.
+
+        Returns:
+            Dictionary of parameters names to the variables.
+        """
+
+        layer.register_parameter(f"{name}_{PTQ_THRESHOLD}",
+                                 nn.Parameter(torch.tensor(self.threshold_values, requires_grad=False)
+                                              if not self.per_channel
+                                              else to_torch_tensor(self.threshold_values),requires_grad=False))
+        layer.register_parameter(f"{name}_{AUXVAR}", nn.Parameter(to_torch_tensor(torch.zeros(self.threshold_shape)),
+                                                                  requires_grad=True))
+
+        # save the quantizer added parameters for later calculations
+        self.quantizer_parameters = {PTQ_THRESHOLD: layer.get_parameter(f"{name}_{PTQ_THRESHOLD}"),
+                                     AUXVAR: layer.get_parameter(f"{name}_{AUXVAR}")}
+
+        return self.quantizer_parameters
+
+
+    def get_aux_variable(self) -> List[torch.Tensor]:
```

Comment:
Ageneral thought: aux variable is very specific to this quantizer. why not call this function get_trainable_variables?

Reply:
It seems very specific to this quantizer now, because we don't have many quantizers implemented, but it is a common thing in many quantizers that we might want to implement in the future (it was part of the gumbel quantizer that we removed, and a uniform quantizer that we also removed for now).
I doesn't necessarilu return all trainable parameters so I wouldn't change the name to this.